modules/common/Utils.py

```diff
@@ -88,3 +108,48 @@ def resource_for_stage(resource):
         resource_stage.uri = resource.uri
         resource_stage.output_filename = os.path.basename(resource.uri)
         return resource_stage
+
+
+class CustomSubProcException(Exception):
+    """Custom subprocess exception"""
+
+
+def subproc(cmd: str, **kwargs) -> None:
+    """Run a cmd as a subprocess. This catches errors that are 'raised' in
+    the STDERR even if the command doesn't return a code > 0.
+
+    Arguments:
+        cmd -- command string to run
+
+    Raises:
+        CustomSubProcException
+    """
+    def raise_exception(sp: subprocess.CompletedProcess, error: str) -> None:
+        err_msg = (f"FAILED to run command '{sp.args}', "
+                   f"with return code '{sp.returncode}', "
+                   f"error: '{error}'")
+        raise CustomSubProcException(err_msg)
+    try:
+        sp = subprocess.run(
+            cmd, shell=True, text=True, capture_output=True, check=False, **kwargs
+        )
+        sp.check_returncode()
+    except subprocess.CalledProcessError as err:
+        raise_exception(sp=sp, error=err)
+    error_keywords = ("error", "killed")
+    if any(e in sp.stderr.lower() for e in error_keywords):
+        raise_exception(sp=sp, error=sp.stderr)
+
+
+def random_temp_file_path(par_dir: Path, ext: str = ".tmp", label_length: int = 8) -> Path:
+    """Create a random temp file path like:
+    <dir>/<random_id>.<ext>
+
+    Arguments:
+        par_dir -- parent directory
```

Comment:
I would call this parameter *path_folder*
Then I would add here a refactored version of *label_length* into *file_name_length*, as it's not a label, but the very file name.
Extensions are meaningless in UNIX like systems, so you don't really need one, and you don't really need to make it a parameter of this method, unless you'd like to expose that functionality to the external world, but I can't think of a case where a client of this _helper method_ would like to have a custom file extension, in which case, 'ext' parameter should be refactored into 'file_extension', and the '.' should not be part of the expected value, as the '.' is what separates the file name from the file estension
> file_name '.' extension

Anything after the last '.' is the file extension.

If we'd like to make this even better, we'd emulate _mktemp_, a call available in shell scripting, by defining this method as
```
def mk_temp_file_path(path_folder):
    # Iterate over creating random file names (uuid is fine) at path.join(path_folder, random_file_name) while file exists
    return random_file_name
```

Instead of a _Path_ I'd probably return just a string representation of the temporary file path, but that's a personal choice and it depends on how the result is being used.

I'm curious about why you decided to use _pathlib_ instead of the _os_ module that was already imported in the script